multiqc/modules/deeptools/bamPEFragmentSize.py

```diff
@@ -0,0 +1,173 @@
+
+            # Read length plot
+            config = {'data_labels': [dict(name="Read length distribution", title="Read length distribution", ylab="Read length (bases)"),
+                                      dict(name="Fragment length distribution", title="Fragment length distribution", ylab="Fragment length (bases)")],
+                      'ylab': "Read length (bases)",
+                      'xlab': "Percentile"}
+            SE = dict()
+            PE = dict()
+            for k, v in self.deeptools_bamPEFragmentSize.items():
+                SE[k] = {0: v['Read Length Minimum'],
+                         10: v['R10'],
+                         20: v['R20'],
+                         25: v['Read Length 1st Quartile'],
+                         30: v['R30'],
+                         40: v['R40'],
+                         50: v['Read Length Median'],
+                         60: v['R60'],
+                         70: v['R70'],
+                         75: v['Read Length 3rd Quartile'],
+                         80: v['R80'],
+                         90: v['R90'],
+                         99: v['R99'],
+                         100: v['Read Length Maximum']}
+                if 'Fragment Length Minimum' not in v:
+                    continue
+                PE[k] = {0: v['Fragment Length Minimum'],
+                         10: v['F10'],
+                         20: v['F20'],
+                         25: v['Fragment Length 1st Quartile'],
+                         30: v['F30'],
+                         40: v['F40'],
+                         50: v['Fragment Length Median'],
+                         60: v['F60'],
+                         70: v['F70'],
+                         75: v['Fragment Length 3rd Quartile'],
+                         80: v['F80'],
+                         90: v['F90'],
+                         99: v['F99'],
+                         100: v['Fragment Length Maximum']}
+            self.add_section(name="bamPEFragmentSize (read/fragment length distribution)",
+                             anchor="bamPEFragmentSize",
+                             plot=linegraph.plot([SE, PE], config))
+
+        return len(self.deeptools_bamPEFragmentSize)
+
+    def parseBamPEFile(self, f, fname):
+        d = {}
+        firstLine = True
+        for line in f.splitlines():
+            if firstLine:
+                firstLine = False
+                continue
+            cols = line.strip().split("\t")
+
+            if len(cols) != 37:
+                # This is not really the output from bamPEFragmentSize!
+                log.warning("{} was initially flagged as the tabular output from bamPEFragmentSize, but that seems to not be the case. Skipping...".format(fname))
+                return dict()
+
+            if cols[0] in d:
+                log.warning("Replacing duplicate sample {}.".format(cols[0]))
+            d[cols[0]] = dict()
```

Comment:
The label in the file is either some user provided name or the relative path to a file given from where ever the tool was executed. I'm hesitant to strip the path, since I know sometimes people use sample names for folders and then have files with the same name in each.

Reply:
Ah, there's a `--fullnames` option. Ignore what I wrote, you've obviously considered that :)

Reply:
Yes sorry, I should have added more context. There's `--fullnames` to prevent this and also [`--dirs`](http://multiqc.info/docs/#sample-names-prefixed-with-directories) to add back on the directory paths. That's what I was referring to above with using the trimmed directory name instead of the default `f['root']` of the log directory. See also the docs for [`clean_s_name`](http://multiqc.info/docs/#custom-sample-names).